swiping changes

ewof opened the pull request:
keybinds added to right/left swieps with ArrowRight and ArrowLeft
made icons a little bigger so they're easier to click on mobile
the flashing dots were moved into an empty message
![image](https://user-images.githubusercontent.com/26354814/228087725-58490494-5148-4d2a-a34f-ab545009607f.png)
the right chevron is there when there are no retires. when a user swipes right if there are no swipes or if the current swipe is the latest swipe it will retry the message again. swiping right when the current swipe is the latest swipe is the same as clicking the retry button
![image](https://user-images.githubusercontent.com/26354814/227821736-75ba3c71-f1d1-4b3c-80a9-ac79320c1ee8.png)
swiping is similar to how it is on tavern/cai where whenever a new retry is added the current swipe is set to the current swipe + 1 (which is the last in the list of retries) and the user just swipes left/right in the direction they want to go
so i have no retries, and i gen a new response with either the refresh button or swiping left, this is what i will see
![image](https://user-images.githubusercontent.com/26354814/227821801-29f46fae-a0e8-49f8-ac1d-d6d8ee2e0ff0.png)
left arrow will bring me to the bots original response, right arrow will gen a new response, when the new (3rd) response is generated i will see 3/3 on the bar and the text will show that of the latest (3rd) response, etc.

sceuick commented:
Cool, I'll check it out today.

ewof commented:
idk why git auto closed the pr when i discarded some commits to merge with upstream but it's open again

sceuick commented on web/pages/Chat/ChatDetail.tsx:
```diff
@@ -263,22 +249,20 @@ const SwipeMessage: Component<{
         </div>
-        <div class="text-[var(--text-800)]">
-          {props.pos + 1} / {props.list.length}
+        <div class={`${props.list.length > 1 ? "visible" : "invisible"} text-white/40`}>
```
I've added a light theme.
`text-900` is the most visible to `text-100` being the least visible.
So here you'd want something like `text-[var(--text-500)]` or `text-500`

sceuick commented on web/pages/Chat/ChatDetail.tsx:
```diff
@@ -263,22 +249,20 @@ const SwipeMessage: Component<{
+          {props.pos} / {props.list.length}
         </div>
-        <div class="cursor:pointer hover:text-[var(--text-800)]">
+        <div class="cursor:pointer hover:text-white">
```
This would need to be `text-900`

sceuick commented on web/store/message.ts:
```diff
@@ -112,14 +113,14 @@ export const msgStore = createStore<MsgState>('messages', {
-      const list = retries[msgId]
-      list.splice(position, 1)
-      const next = [replacement].concat(list)
+      // const list = retries[msgId]
```
So this code would move the confirmed swipe to the front of the list and the current swipe position would be set to `0`.
Is the idea now that the order never changes?

I figured in a flow that looks like this:

- Bot responds
- Retry, retry, retry, retry (swipes now shows 1/5)
- Swipe to 3/5
- Confirm
- User responds
- Bot responds
- User deletes bot response + user response
- Swipe position is back to `0` which is wrong, it should be `2` (which represents 3/5)

ewof replied:
>Is the idea now that the order never changes?

yea, instead of confirming with a checkmark the user confirms when they respond to the response they picked by swiping

>User deletes bot response + user response
Swipe position is back to `0` which is wrong, it should be `2` (which represents 3/5)

with latest commit when a new response is generated the swipe is set to the latest one. when a user confirms by responding to their selected response, if they delete the new response to go back to the old message swipe is set the the last swipe in that messages retires

sceuick commented on web/store/message.ts:
```diff
@@ -6,6 +6,7 @@ import { local } from './data/storage'
 import { subscribe } from './socket'
 import { toastStore } from './toasts'
 import { userStore } from './user'
+import { setSwipe } from '../pages/Chat/ChatDetail'
```
A store shouldn't import from a component. This function should be passed as a parameter to the function instead.

sceuick commented on web/pages/Chat/ChatDetail.tsx:
```diff
@@ -57,17 +51,29 @@ const ChatDetail: Component = () => {
     setSwipe(next)
   }
 
+  window.onkeydown = (e: KeyboardEvent): any => {
```
This behaviour will need to be done in `onMount` then removed when navigating away using `onCleanup`

ewof committed:
merge upstream and make it right-focused again

sceuick committed:
Chat export (#95)

- Move chat options into modal
- Mobile styling improvements
- Fix openai max context bug
- Fix ios input bar below fold

khanonnie committed:
Add preset duplication (#84)

Co-authored-by: Sceuick <leikho@gmail.com>

ewof committed:
show gaslight prompt if using (#90)

- Fix sending highlighted text
- Only use host settings for group chat
- Only clear input text if message successfully sent

Co-authored-by: Sceuick <leikho@gmail.com>

sceuick committed:
Fix api build env vars [skip ci]

ewof committed:
merge

ewof committed:
Merge branch 'dev' into swiping-changes

ewof committed:
Merge branch 'dev' into swiping-changes

ewof committed:
Merge branch 'luminai-companion:dev' into swiping-changes

ewof committed:
Merge branch 'luminai-companion:dev' into swiping-changes
